Shopify/dawn
# Enable independent corner radii in grid items

**Why are these changes introduced?**

Refs https://github.com/Shopify/dawn/issues/922

This PR, in preparation for the border/corner collapse work, _allows_ all 4 of the corner radii of cards, media, and content containers _within grid layouts_ to be set and overridden independently. 

Note: This is a change in implementation only. No visual or functional changes were made.

**What approach did you take?**
In addition to the existing `--border-radius` vars, I'm also implementing corner-specific variables for each corner (e.g.`--top-left-radius`) which exist in the `.contains-<element type> .grid__item` context. For now, this is just an overcomplicated way to get us to the same place we already are, in that all 4 corners will always have identical values. Items not in the applicable grid context will continue to use the existing `border-radius: var(--<element type>-radius)` fallback regardless.

**Other considerations**

**Testing steps/scenarios**
1. Most importantly we want to ensure no corner radius related regressions were introduced. To do so, apply corner radius, border, and shadow to media, content containers, and cards via global settings and briefly check the following..
- [ ] featured collection
- [ ] main product collection
- [ ] search results
- [ ] featured blog
- [ ] main blog
- [ ] multicolumn
- [ ] collection list
- [ ] product media (thumbnails and stacked layouts, images and deferred media content, focus outlines)

2. Additionally to verify the updates are working as intended, you should be able to manually add corner specific vars via the dev inspector and see the result in applicable grids.
<img src="https://screenshot.click/25-36-0ga9z-wdca1.png" width="500" />

**Demo links**

- [Editor](https://os2-demo.myshopify.com/admin/themes/127647318038/editor)

**Checklist**
